### PR TITLE
 GEOMESA-2642,GEOMESA-2562 Upgrade to GeoTools 21.1 and GeoServer 2.15.1

### PR DESCRIPTION
**Pull request by joshfix**
Updated dependency versions and performed slight refactoring to account for new GeoTools packages.

Signed-off-by: Josh Fix <josh@federal.planet.com>

### CODE REVIEW

**Comment by elahrvivaz** [comment]
@joshfix I pushed some changes to your branch to clean up the poms a little and to add the cqs list. I'm going to open CQs with Eclipse for the new jars - we'll have to wait for approval, then we can merge. thanks again for getting it working!

File: build/cqs.tsv [hunk 1]
```diff
@@ -170,47 +171,46 @@ org.calrissian.mango:mango-core	1.2.0	compile
...
-org.geotools.ogc:net.opengis.wfs	20.0	compile
-org.geotools.ogc:org.w3.xlink	20.0	compile
-org.geotools.xsd:gt-xsd-core	20.0	compile
-org.geotools.xsd:gt-xsd-fes	20.0	compile
-org.geotools.xsd:gt-xsd-filter	20.0	compile
-org.geotools.xsd:gt-xsd-gml2	20.0	compile
-org.geotools.xsd:gt-xsd-gml3	20.0	compile
-org.geotools.xsd:gt-xsd-ows	20.0	compile
-org.geotools.xsd:gt-xsd-wfs	20.0	compile
+org.geotools:gt-coverage	21.1	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: geotools: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20102

File: build/cqs.tsv [hunk 2]
```diff
@@ -40,6 +40,7 @@ commons-dbcp:commons-dbcp	1.4	compile
 commons-digester:commons-digester	1.8	compile
 commons-httpclient:commons-httpclient	3.1	compile
 commons-io:commons-io	2.5	compile
+commons-jxpath:commons-jxpath	1.3	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20103

File: build/cqs.tsv [hunk 3]
```diff
@@ -78,38 +79,38 @@ io.dropwizard.metrics:metrics-graphite	3.1.2	compile
...
+it.geosolutions.jaiext.rlookup:jt-rlookup	1.1.9	compile
+it.geosolutions.jaiext.scale2:jt-scale2	1.1.9	compile
+it.geosolutions.jaiext.scale:jt-scale	1.1.9	compile
+it.geosolutions.jaiext.shadedrelief:jt-shadedrelief	1.1.9	compile
+it.geosolutions.jaiext.stats:jt-stats	1.1.9	compile
+it.geosolutions.jaiext.translate:jt-translate	1.1.9	compile
+it.geosolutions.jaiext.utilities:jt-utilities	1.1.9	compile
+it.geosolutions.jaiext.vectorbin:jt-vectorbin	1.1.9	compile
+it.geosolutions.jaiext.warp:jt-warp	1.1.9	compile
+it.geosolutions.jaiext.zonal:jt-zonal	1.1.9	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: jai: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20105

File: build/cqs.tsv [hunk 4]
```diff
@@ -170,47 +171,46 @@ org.calrissian.mango:mango-core	1.2.0	compile
 org.codehaus.jackson:jackson-core-asl	1.9.3	compile
 org.codehaus.jackson:jackson-jaxrs	1.9.3	compile
 org.codehaus.jackson:jackson-mapper-asl	1.9.3	compile
-org.eclipse.emf:org.eclipse.emf.common	2.12.0	compile
-org.eclipse.emf:org.eclipse.emf.ecore	2.12.0	compile
-org.eclipse.emf:org.eclipse.emf.ecore.xmi	2.12.0	compile
+org.eclipse.emf:org.eclipse.emf.common	2.15.0	compile
+org.eclipse.emf:org.eclipse.emf.ecore	2.15.0	compile
+org.eclipse.emf:org.eclipse.emf.ecore.xmi	2.15.0	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: eclipse projects don't require a cq, see https://dev.eclipse.org/ipzilla/show_bug.cgi?id=8723

File: build/cqs.tsv [hunk 5]
```diff
@@ -170,47 +171,46 @@ org.calrissian.mango:mango-core	1.2.0	compile
...
+org.geotools.ogc:org.w3.xlink	21.1	compile
+org.geotools.xsd:gt-xsd-core	21.1	compile
+org.geotools.xsd:gt-xsd-fes	21.1	compile
+org.geotools.xsd:gt-xsd-filter	21.1	compile
+org.geotools.xsd:gt-xsd-gml2	21.1	compile
+org.geotools.xsd:gt-xsd-gml3	21.1	compile
+org.geotools.xsd:gt-xsd-ows	21.1	compile
+org.geotools.xsd:gt-xsd-wfs	21.1	compile
 org.hsqldb:hsqldb	2.3.0	compile
+org.huldra.math:bigint	0.7.1	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: pending... having some trouble tracking down the source code.

Comment (jnh5y) [reply build/cqs.tsv]: @elahrvivaz what pulled this in!?

Comment (elahrvivaz) [reply build/cqs.tsv]: Andrea Aime added it to jai-ext to speed up some operations I guess. I was able to find the source code by posting on their mailing list, so I'll have a cq up soon.

Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20114

**Comment by elahrvivaz** [comment]
force pushed to fix the merge

**Comment by elahrvivaz** [comment]
@jnh5y could you ping the pmc for:
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20102
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20103
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20105
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20114 (hasn't ticked over to pmc status yet, but should in the next hour or so)
Thanks!

**Comment by jnh5y** [comment]
> @jnh5y could you ping the pmc for:
> https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20102
> https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20103
> https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20105
> https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20114
> edit: all are in pmc status now ~(hasn't ticked over to pmc status yet, but should in the next hour or so)~
> Thanks!

I've emailed the PMC.  Should take a few days, etc, etc.

File: build/cqs.tsv (hunk 2 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: approved

File: build/cqs.tsv (hunk 3 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: approved

**Comment by joshfix** [comment]
Is there any special coordination that needs to occur with this PR if I also want to submit a PR for updating the geoserver plugin to the same versions?  Or can that be handled completely independently?

**Comment by elahrvivaz** [comment]
@joshfix there's no official coordination needed - feel free to put up a corresponding PR against geomesa-geoserver, which builds against this snapshot. We won't be able to merge it until this PR is merged, but we can review/comment/test in the meantime. Generally, with transitive PRs, I just add a comment back to the original so that people are aware. Thanks,

File: build/cqs.tsv (hunk 1 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: approved

File: build/cqs.tsv (hunk 5 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: approved

**Comment by elahrvivaz** [comment]
@jnh5y any last objections or should i push the button?

**Pull request merged**